### DISCUSSION TITLE
Bridge Working Group Update [2021-04-02]

### DISCUSSION THREAD

**Discussion by alexauroradev** [aurora-is-near/rainbow-bridge#549] [General]
The video of the call is available [here](https://www.youtube.com/watch?v=L2rFzsXb1Ck)
The forum discussion is available [here](https://gov.near.org/t/bridge-working-group-update-2021-04-02/1127)

## Agenda
- Status, current goal
- Updates
- Discussion

## Status, current goal
Bridge is on the mainnet, we're fixing minor bugs and preparing for the next quarter

### Alex Shevchenko
- EVM stuff (OKRs, testnet harfork, general management)
- eth-connector consulting
- near-erc20-connector consulting & review
- Bridge PR & misc

### Marcelo Fornet
- Relayer (issues with 750 sec, metrics)
- Testnet hardfork issues:
    - Funding of the relayer: consult with Mario
    - Wait till tomorrow (protocol version upgrade 2 epoch)
- Problem with enough attached deposit

### Eugene Kapun
- Found ethash bug (u32 overflow)
    - We now have 2 tools with coinsiding results for calculating DAG roots
- EVM Math API review
- Hiring

### Kirill Abramov
- Ethproof bugfix PR is prepared
- Eth client code is updated, tests added
- Testing and debugging `no-std` version of eth-connector
- Update of the README of the eth-connector in the PR

### Pierre-Alain Ouvrard
- UI updates with Matt and Chad
- Debugging and fixing WalletConnect
    - Requires more testing
- Merkle proofs issue
    - Status field in the receipt is non-consistent because of the web3-provider library of WalletConnect
- Browser support issue (Firefox, Safari)
    - Both of the browsers cannot load the MPT library
    - We need to find a way how to make it work
    - ethereum.js is aware of the bundling issue

### Matt Henderson
- Review of the front-end

## Discussion
- Update bridge contracts during the next week?
    - Add interface for updating DAG merkle roots?
- Ropsten<>Testnet bridge should be fixed
- Browser issue: do the check of the browser before loading the dependency
- We need to prepara FE for bridging ETH and $NEAR
    - Bridge ETH where? NEAR native / EVM

**Comment by maxzaver** [comment]
It is unclear from this updates what things were done vs things that are still in progress.
